### ISSUE TITLE
An issue about let statement in typescript.

### ISSUE THREAD

**Issue by arliang** [Microsoft/TypeScript#6356]
http://www.typescriptlang.org/Playground#src=for(let%20i%20%3D%200%3B%20i%20%3C%2010%3B%20i%2B%2B)%7B%0D%0A%09setTimeout(function()%7B%0D%0A%09%09console.log(i)%3B%0D%0A%09%7D)%3B%0D%0A%7D

```typesceipt
for(let i = 0; i < 3; i++){
	setTimeout(function(){
		console.log(i);
	});
}
```

```javacript
for (var i = 0; i < 3; i++) {
    setTimeout(function () {
        console.log(i);
    });
}
```

In es6, the first example outputs `0 1 2`
But the JS which translated by TS outputs `3 3 3`

It would act like es6 if wrapped with `(function(i){ /*  */ })(i)`
```javacript
for (var i = 0; i < 10; i++) {
    (function(i){
        setTimeout(function () {
            console.log(i);
        });
    })(i);
}
```

**Comment by RyanCavanaugh** [comment]
This has already been implemented in `f0ba3f47`

**Comment by RyanCavanaugh** [comment]
Duplicate #3915
